dhis-2/dhis-api/src/main/java/org/hisp/dhis/calendar/DateUnitPeriodTypeParser.java

```diff
@@ -204,6 +204,37 @@ else if ( DateUnitType.QUARTERLY == dateUnitType )
 
             return new DateInterval( start, end );
         }
+        else if ( DateUnitType.QUARTERLY_NOV == dateUnitType )
+        {
+            int year = Integer.parseInt( matcher.group( 1 ) );
```

Comment:
I believe that are too many "magic numbers" from lines 209 to 231.
Probably giving a name to them would make things easier to read/understand.

Reply:
have put a doc explaining the logic